### ISSUE TITLE
[🐛 Bug]: PATH not properly read.

### ISSUE THREAD

**Issue by AliSot2000** [SeleniumHQ/selenium#10781]
### Selenium failed to find the geckodriver executable inside the `$PATH` variable. 

I tried a custom entry in the path, pointing to a folder in my home directory `/home/alisot2000/.geckodriver`, I tried `/usr/local/bin` and `/usr/bin`.

The GeckoDriver works perfectly fine, if provided as an argument when instantiating `webdriver.Firefox(executable_path="/path/to/geckodriver")`.

I went through the code executed with the debugger of Pycharm and it didn't seem to "load" the Environment variable of the OS. It just used the `DEFAULT_EXECUTABLE_PATH = "geckodriver"` inside the firefox/service.py file as the `executable_path`. I also tried to put only the parent directory of the geckodriver in the `$PATH` variable as well as putting the path to the executable directly into the `$PATH` variable, both with equal result. 

I read the file `common/service.py` containing the follow part:
```python
    def start(self):
        """
        Starts the Service.

        :Exceptions:
         - WebDriverException : Raised either when it can't start the service
           or when it can't connect to the service
        """
        try:
            cmd = [self.path]
            cmd.extend(self.command_line_args())
            self.process = subprocess.Popen(cmd, env=self.env, # <-- !!!
                                            close_fds=system() != 'Windows',
                                            stdout=self.log_file,
                                            stderr=self.log_file,
                                            stdin=PIPE,
                                            creationflags=self.creationflags)

```

There (`<-- !!!`) I verified with the debugger that the `self.env` contains the `$PATH` variable which points to the geckoddriver executable. As I mentioned, the subprocess fails to start. 

I guess that this is a more general issue since it should have worked in the past, judging by [forum posts](https://askubuntu.com/questions/870530/how-to-install-geckodriver-in-ubuntu) in the past. It seems that this is a bug since it works if the `executable_path` (and subsequently the `cmd` in the start method) is actually the absolute path to the executable.

### How can we reproduce the issue?

```shell
This produces the error:


from selenium import webdriver

driver = webdriver.Firefox()
driver.get("https://www.google.com/")
```

### Relevant log output

```shell
Traceback (most recent call last):
  File "/home/alisot2000/project/lin_venv/lib/python3.8/site-packages/selenium/webdriver/common/service.py", line 71, in start
    self.process = subprocess.Popen(cmd, env=self.env,
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'geckodriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alisot2000/project/python_scripts/Main.py", line 22, in <module>
    driver = webdriver.Firefox()
  File "/home/alisot2000/project/lin_venv/lib/python3.8/site-packages/selenium/webdriver/firefox/webdriver.py", line 175, in __init__
    self.service.start()
  File "/home/alisot2000/project/lin_venv/lib/python3.8/site-packages/selenium/webdriver/common/service.py", line 81, in start
    raise WebDriverException(
selenium.common.exceptions.WebDriverException: Message: 'geckodriver' executable needs to be in PATH.
```

### Operating System

Ubuntu 20.04

### Selenium version

Python 3.8, Selenium 4.1.5

### What are the browser(s) and version(s) where you see this issue?

Firefox 101.0.1 64bit

### What are the browser driver(s) and version(s) where you see this issue?

GeckoDriver_0.31

### Are you using Selenium Grid?

No / Not to my knowledge

**Comment by symonk** [comment]
what does `which geckodriver` say? and is it on `$PATH`.

**Comment by AliSot2000** [comment]
```bash
alisot2000@kubuntu:~$ which geckodriver
alisot2000@kubuntu:~$ which htop
/usr/bin/htop
alisot2000@kubuntu:~$ 
```

**Comment by symonk** [comment]
so `geckodriver` is not on path? for example if you move it to `/usr/local/bin` it will probably work as expected (I know you mentioned that in the issue - could we try an example with that?) - Thanks.  Selenium is simply running something along the lines of:

`geckodriver --websocket-port XXXXX --port XXXXX` when no executable path is provided to the driver directly (now) - soon to be deprecated in favour of using a service mind you)

**Comment by symonk** [comment]
@AliSot2000 Thank you very much for a very detailed report; I think the root cause is that `geckodriver` is not available on the path when running without specifying an `executable_path` explicitly; I will close this one as cannot recreate, if you think this is incorrect, please re-open it and we can discuss, but everything points as no executable on the path

**Comment by AliSot2000** [comment]
Ok, I think I might have figured out the issue. 
I tried to create symlinks because I wanted the file to to be located inside my home folder (because I tend to forget what I have installed). 

I assumed it would work since a `ll` inside the `/usr/bin` would reveal a lot of symlinks to other executables.

Ok. Well I'm sorry for the confusion.

Edit: 

I tested this symlink theory of mine again and came to a very simple solution. I reprated the same thing I did on the main machine and I used some saved commands I had from previously installing the geckodriver. For some reason I thought it was smart to rename the binary. Now it becomes rather obvious why it wasn't found as well as why my path wasn't working.
